slam-libraries/viam-cartographer/src/slam_service/config.cc

```diff
@@ -94,6 +104,23 @@ void ParseAndValidateConfigParams(int argc, char** argv,
     slamService.data_rate_ms = std::chrono::milliseconds(FLAGS_data_rate_ms);
     slamService.map_rate_sec = std::chrono::seconds(FLAGS_map_rate_sec);
 
+    // TODO: Remove once PR #1689 is submitted
+    // https://github.com/viamrobotics/rdk/pull/1689 This will allow intergation
+    // tests to pass (See associated JIRA ticket:
+    // https://viam.atlassian.net/browse/RSDK-1593)
+    if (FLAGS_delete_processed_data.empty()) {
+        slamService.delete_processed_data = false;
```

Comment:
since the current integration tests are using a version of rdk that does not pass the `delete_process_data` flag into cartographer, this if statement combined with the string flag you asked about above is used to effectively ignore the missing input, and will be removed once integration tests pass in rdk.

Reply:
Ok I think I understand - thanks for explaining!